### PR TITLE
Support _Nonnull annotation.

### PR DESCRIPTION
**Pull request by rbehrends**

### CODE REVIEW

File: ward/cparser.lua [hunk 1]
```diff
@@ -138,7 +138,7 @@ local keywords = {
   'int', 'switch', 'double', 'long', 'typedef', 'else', 'register',
   'union', '__asm', '__asm__', '__thread', '__builtin_va_list', '__inline__',
   '__builtin_va_arg', '__extension__', '__const', '__restrict', '__volatile__',
-  '__signed', '__unsigned', '_Alignof',
+  '__signed', '__unsigned', '_Alignof', '_Nonnull',
```
Comment (fingolfin) [inline ward/cparser.lua:141]: So, why is this handled here, but e.g. `_Noreturn` is handled in `config.lua` like this:
```
preprocessor_options = {
  "-D__extension__=",
  "-D__attribute__(x)=",
  -- Ignore C11 annotations.
  "-D_Noreturn=",
  "-D_Alignas(x)=",
  -- TODO: The following is a workaround around __typeof() occurring
  -- in Linux header files. The correct solution is to properly deal
  -- with typeof() and __typeof() expressions.
  "-D__typeof(x)=int"
}
```

Comment (rbehrends) [reply ward/cparser.lua:141]: Ideally, it should be handled in the same fashion, yes.

**Comment by ChrisJefferson** [comment]
Similarly need to handle: `int (* _Nullable _close)(void *);`, which also appears.

File: ward/cparser.lua (hunk 1 above)
Comment (ChrisJefferson) [reply ward/cparser.lua:141]: @rbehrends : to be clear, we should deal with `_Nonnull` as a preprocessor option?

**Comment by fingolfin** [comment]
As a temporary and minimal invasive workaround, we could then add `-D_Nonnull=` to the `preprocessor_options` list.

**Comment by fingolfin** [comment]
This has been superseded by PR #27

**Pull request closed without merging**